src/backend.rs

```diff
@@ -193,6 +196,11 @@ impl Mock {
 	pub fn set_read_timeout_error(&mut self, err: Option<io::Error>) {
 		self.set_read_timeout_error = err;
 	}
+
+	/// Set the callback function to generate custom replies.
+	pub fn set_reply_callback(&mut self, callback: fn(&[u8]) -> Vec<u8>) {
```

Comment:
I really like the flexibility this feature adds! However, I would prefer that the callback not allocate a `Vec` needlessly, as the mock will just copy it into it's own buffer. This could be easily done by changing the signature of the callback to the following, which would allow the callback to write directly into the Mock's reply buffer.

```rust
callback: fn(&[u8], &mut dyn io::Write)
```